libs/langchain/langchain/vectorstores/neo4j_vector.py

```diff
@@ -0,0 +1,623 @@
+            "$embedding_node_property,"
+            "toInteger($embedding_dimension),"
+            "$similarity_metric )"
+        )
+
+        parameters = {
+            "index_name": self.index_name,
+            "node_label": self.node_label,
+            "embedding_node_property": self.embedding_node_property,
+            "embedding_dimension": self.embedding_dimension,
+            "similarity_metric": self._distance_strategy,
+        }
+        self.query(index_query, parameters)
+
+    @property
+    def embeddings(self) -> Embeddings:
+        return self.embedding_function
+
+    @classmethod
+    def __from(
+        cls,
+        texts: List[str],
+        embeddings: List[List[float]],
+        embedding: Embeddings,
+        username: str,
+        password: str,
+        url: str,
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Neo4jVector:
+        if ids is None:
+            ids = [str(uuid.uuid1()) for _ in texts]
+
+        if not metadatas:
+            metadatas = [{} for _ in texts]
+
+        store = cls(
+            username=username,
+            password=password,
+            url=url,
+            embedding_function=embedding,
+            **kwargs,
+        )
+
+        # Check if the index already exists
+        embedding_dimension = store.retrieve_existing_index()
+
+        # If the index doesn't exist yet
+        if not embedding_dimension:
+            store.create_new_index()
+        # If the index already exists, check if embedding dimensions match
+        elif not store.embedding_dimension == embedding_dimension:
+            raise ValueError(
+                f"Index with name {store.index_name} already exists."
+                "The provided embedding function and vector index "
+                "dimensions do not match.\n"
+                f"Embedding function dimension: {store.embedding_dimension}\n"
+                f"Vector index dimension: {embedding_dimension}"
+            )
+
+        store.add_embeddings(
+            texts=texts, embeddings=embeddings, metadatas=metadatas, ids=ids, **kwargs
+        )
+
+        return store
+
+    def add_embeddings(
+        self,
+        texts: Iterable[str],
+        embeddings: List[List[float]],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Add embeddings to the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            embeddings: List of list of embedding vectors.
+            metadatas: List of metadatas associated with the texts.
+            kwargs: vectorstore specific parameters
+        """
+        if ids is None:
+            ids = [str(uuid.uuid1()) for _ in texts]
+
+        if not metadatas:
+            metadatas = [{} for _ in texts]
+
+        import_query = (
+            "UNWIND $data AS row "
```

Comment:
let's wrap this in `CALL { WITH row  CREATE/MERGE } IN TRANSACTIONS OF 1000 ROWS`
or batch from the client side as creating the data esp. with the big embeddings can blow things up.

Reply:
if id's are provided externally (i.e. non-generated, we should use MERGE)

otherwise we can also us the cypher function `randomUUID()` and return the new ids from the query.